packages/forms/src/index.tsx

```diff
@@ -150,68 +152,249 @@ const useErrorStyles = ({
   return { className, style }
 }
 
-const valueAsProps = {
-  valueAsBoolean: (value: string) => !!value,
-  valueAsJSON: (value: string) => {
-    try {
-      return JSON.parse(value)
-    } catch (e) {
-      return null
-    }
+// Used to determine if a value is empty.
+const isValueEmpty = (val: string): boolean => val === ''
+
+/**
+ * EmptyAsValue defines the values that can be used for the field emptyAs prop
+ * It sets the value to be returned from the field if the field is empty.
+ * If the valueOf prop is truly undefined (not 'undefined'), it will return
+ * a default value corresponding to the type of field.  (See
+ * the comments above the setCoercion function for more details)
+ */
+
+type EmptyAsValue = null | 'undefined' | 0 | ''
```

Comment:
Should this one include `'NaN'` as well?

Reply:
I debated this one and had at one point started adding it. However, I pulled back the `NaN` option as I couldn't think of a case where I would want to do this in a real application.  If you want to add it, I'm not opposed to it.  
Just let me know.

Reply:
I agree. Can't see where that would be useful. But I did notice it in some tests. That's why I asked 🙂

Reply:
The code right now has a `<NumberField>` return a NaN on an empty case, if none of the other conditions apply (as described in the docs).  This was to replicate existing behavior, just in case someone was relying on this.   I believe this is ok, but let me know if you think otherwise.

Reply:
Returning `NaN` for `<NumberField>`s is fine. And having tests for that is great 🙂  I was asking about the tests that try to set `valueAs` to `'NaN'` for `<TextField>`s. I made a comment about it further up.